### PR TITLE
[dev] Implement server-side bits for the State/SetState uniter API endpoints

### PR DESCRIPTION
**Pull request by achilleasa**
## Description of change

This PR is part of the work required for supporting server-side storage of charm state. 

It builds on the changes introduced by #11136 and adds the server-side logic for handling the new SetState/State calls for the uniter API

## QA steps

QA steps not available as the client-side bits are not currently wired to call out to this facade.

### CODE REVIEW

File: apiserver/facades/agent/uniter/uniter.go [hunk 1]
```diff
@@ -3164,3 +3181,69 @@ func (u *UniterAPI) updateUnitNetworkInfo(unitTag names.UnitTag) error {
 
 	return settingsGroup.Write()
 }
+
+func (u *UniterAPI) State(args params.Entities) (params.UnitStateResults, error) {
```
Comment (SimonRichardson) [inline apiserver/facades/agent/uniter/uniter.go]: Nit: comment.

File: apiserver/facades/agent/uniter/uniter.go [hunk 2]
```diff
@@ -3164,3 +3181,69 @@ func (u *UniterAPI) updateUnitNetworkInfo(unitTag names.UnitTag) error {
...
+
+		if res[i].State, err = unit.State(); err != nil {
+			res[i].Error = common.ServerError(err)
+		}
+	}
+
+	return params.UnitStateResults{Results: res}, nil
+}
+
+func (u *UniterAPI) SetState(args params.SetUnitStateArgs) (params.ErrorResults, error) {
```
Comment (SimonRichardson) [inline apiserver/facades/agent/uniter/uniter.go]: Nit: comment.

File: apiserver/facades/agent/uniter/uniter.go [hunk 3]
```diff
@@ -3099,7 +3116,7 @@ func (u *UniterAPI) UpdateNetworkInfo(args params.Entities) (params.ErrorResults
 		}
 
 		if !canAccess(unitTag) {
-			res[i].Error = common.ServerError(err)
+			res[i].Error = common.ServerError(common.ErrPerm)
```
Comment (achilleasa) [inline apiserver/facades/agent/uniter/uniter.go]: Drive-by fix.

File: apiserver/facades/agent/uniter/uniter.go (hunk 1 above)
Comment (achilleasa) [reply apiserver/facades/agent/uniter/uniter.go]: Doh! How did I forget the comments?

**Comment by achilleasa** [comment]
$$merge$$

**Comment by achilleasa** [comment]
$$merge$$

(failed due to unrelated WorkerSuite.TestContainerStart)
